sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1988,4 +1988,51 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       assert(errMsg.startsWith("The field for corrupt records must be string type and nullable"))
     }
   }
+
+  test("SPARK-18772: Parse special floats correctly") {
+    val jsons = Seq(
+      """{"a": "+INF"}""",
+      """{"a": "INF"}""",
+      """{"a": "-INF"}""",
+      """{"a": "NaN"}""",
+      """{"a": "Infinity"}""",
+      """{"a": "+Infinity"}""",
+      """{"a": "-Infinity"}""")
```

Comment:
I do not understand which standards we follow. The above seven values are accepted by `toFloat` and `toDouble`?